docs/gallery/plot_external_resources.py

```diff
@@ -306,6 +307,52 @@ def __init__(self, **kwargs):
     entity_uri='https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=10090'
 )
 
+###############################################################################
+# Using add_ref_termset
+# ------------------------------------------------------
+# :py:class:`~hdmf.common.resources.HERD` has multiple ways for users to add
+# external references. The :py:func:`~hdmf.common.resources.HERD.add_ref_termset`
+# method allows users to not only validate terms, i.e., keys, but also adds the
+# ability to iteratively add references for entire datasets, lists, arrays, etc.
+
+# The :py:func:`~hdmf.common.resources.HERD.add_ref_container` method is directly
+# used for populating :py:class:`~hdmf.common.resources.HERD` when writing a file.
+# :py:func:`~hdmf.common.resources.HERD.add_ref_termset` can be used for new files;
+# however, it is also the best practice when adding references for existing files.
```

Comment:
I'm a bit confused here. When should a user use `add_ref_container`? Or should that be used only internally?

Reply:
As far as I can tell, I think `add_ref_container` in the code is right now used only internally. To use the function with existing files, it seems a user would need to wrap existing data fields with TermSetWrapper and then call `add_ref_container`. The alternative, would be to use `add_ref_termset` for existing files. In principle, I think either way could work, but I'm not sure if it's currently possible to wrap data fields that have been read from disk with TermSetWrapper (but maybe that's possible, not sure).

Reply:
I'm not sure if/when a user would need to call ``add_ref_container`` directly.